reference/commands/install.rst

```diff
@@ -1,2 +1,273 @@
+"build" and "host" context. That means that if a profile with the "default" name has not been created, it will error.
+
+Multiple definitions of profiles can be passed as arguments, and they will compound from left to right (right has the
+highest priority)
+
+.. code-block:: bash
+
+    # The values of myprofile 3 will have higher priority
+    $ conan install . -pr=myprofile1 -pr=myprofile2 -pr=myprofile3
+
+If values of ``settings``, ``options`` and ``conf`` are provided in the command line, they create a profile that
+is composed with the other provided ``-pr`` (or the "default" one if not specified) profiles, with higher priority,
+not matter what is the order of arguments).
+
+.. code-block:: bash
+
+    # the final "host" profile will always be build_type=Debug, even if "myprofile"
+    # says "build_type=Release"
+    $ conan install . -pr=myprofile -s build_type=Debug
+    
+
+Generators and deployers
+------------------------
+
+The ``-g`` argument allows to define in command line the different built-in generators to be used:
+
+.. code-block:: bash
+
+    $ conan install --requires=zlib/1.2.13 -g CMakeDeps -g CMakeToolchain
+
+Note that in the general case, the recommended approach is to have the ``generators`` defined in the ``conanfile``, 
+and only for the ``--requires`` use case, it would be more necessary as command line argument.
+
+Generators are intended to create files for the build systems to locate the dependencies, while the ``deployers``
+main use case is copy files from the Conan cache to user space, and other custom operations over the dependency graph,
+like collecting licenses, generating reports, deploying binaries to the system, etc. The syntax for deployers is:
+
+.. code-block:: bash
+
+    # does a full copy of the dependencies binaries to the current user folder
+    $ conan install . --deploy=full_deploy
+
+
+There are 2 built-in deployers:
+
+- ``full_deploy`` does a complete copy of the dependencies binaries in the local folder, with a minimal folder
+  structure to avoid conflicts between files and artifacts of different packages
+- ``direct_deploy`` does a copy of only the immediate direct dependencies, but does not include the transitive
+  dependencies.
+
+
+Some generators might have the capability of redefining the target "package folder". That means that if some other
+generator like ``CMakeDeps`` is used that is pointing to the packages, it will be pointing to the local deployed
+copy, and not to the original packages in the Conan cache.
+
+It is also possible, and it is a powerful extension point, to write user custom deployers.
+Read more about custom deployers in :ref:`reference_extensions_deployers`.
+
+
+Name, version, user, channel
+----------------------------
+
+The ``conan install`` command provides optional arguments for ``--name, --version, --user, --channel``. These 
+arguments might not be necessary in the majority of cases. Never for ``conanfile.txt`` and for ``conanfile.py``
```

Comment:
What does it mean here:
> Never for ``conanfile.txt`` and for ``conanfile.py``

That `--name` etc should not be used if using a `conanfile...`? Or am I misunderstanding the meaning?

Reply:
Never for ``conanfile.txt``.
It only makes sense for ``conanfile.py``